### ISSUE TITLE
Errors in LMDB read and Registering ImageData after adding custom layer 

### ISSUE THREAD

**Issue by serimp** [BVLC/caffe#3579]
Hi,

After adding a custom layer (called 'Topology' layer, followed https://github.com/BVLC/caffe/wiki/Development ), errors appear in registering the ImageData layer and Data layer, regardless of the fact that the custom layer is included or not included. 

- When ImageData layer is used:

Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: ImageData (known types: AbsVal, Accuracy, ArgMax, BNLL, BatchNorm, BatchReindex, Concat, ContrastiveLoss, Convolution, Data, Deconvolution, Dropout, DummyData, Eltwise, Embed, EuclideanLoss, Exp, Filter, Flatten, HDF5Data, HDF5Output, HingeLoss, Im2col, InfogainLoss, InnerProduct, LRN, Log, MVN, MemoryData, MultinomialLogisticLoss, PReLU, Pooling, Power, ReLU, Reduction, Reshape, SPP, Sigmoid, SigmoidCrossEntropyLoss, Silence, Slice, Softmax, SoftmaxWithLoss, Split, TanH, Threshold, Tile, Topology)
*** Check failure stack trace: ***
    @     0x7f6389e18daa  (unknown)
    @     0x7f6389e18ce4  (unknown)
    @     0x7f6389e186e6  (unknown)
    @     0x7f6389e1b687  (unknown)
    @     0x7f638a465ac4  caffe::LayerRegistry<>::CreateLayer()
    @     0x7f638a46bab5  caffe::Net<>::Init()
    @     0x7f638a46dc25  caffe::Net<>::Net()
    @     0x7f638a54a92a  caffe::Solver<>::InitTrainNet()
    @     0x7f638a54ba3c  caffe::Solver<>::Init()
    @     0x7f638a54bd49  caffe::Solver<>::Solver()
    @     0x7f638a4375c3  caffe::Creator_SGDSolver<>()
    @           0x40e71e  caffe::SolverRegistry<>::CreateSolver()
    @           0x40794b  train()
    @           0x405841  main
    @     0x7f6389126ec5  (unknown)
    @           0x405f51  (unknown)
    @              (nil)  (unknown)
./train_image.sh: line 3:   904 Aborted                 (core dumped)

- When Data layer is used with LMDB:

 Unknown database backend
*** Check failure stack trace: ***
    @     0x7f213f882daa  (unknown)
    @     0x7f213f882ce4  (unknown)
    @     0x7f213f8826e6  (unknown)
    @     0x7f213f885687  (unknown)
    @     0x7f213fe8f060  caffe::db::GetDB()
    @     0x7f213ffbd25c  caffe::DataReader::Body::InternalThreadEntry()
    @     0x7f213fe678e0  caffe::InternalThread::entry()
    @     0x7f2138771a4a  (unknown)
    @     0x7f2134f78182  start_thread
    @     0x7f213ec6947d  (unknown)
    @              (nil)  (unknown)
./train_topology.sh: line 5:  2448 Aborted

**Comment by serimp** [comment]
Resolved after setting OPENCV, LMDB flag back to 1 and recompiling.

**Comment by wolfbill** [comment]
Hi serimp,tell me more about how to solve the problem
